torch/sparse/_semi_structured_ops.py

```diff
@@ -166,3 +169,26 @@ def semi_sparse_linear(func, types, args=(), kwargs=None) -> torch.Tensor:
         )
 
     return res.view(*shape[:-1], -1)
+
+
+def semi_sparse_scaled_mm(func, types, args=(), kwargs=None) -> torch.Tensor:
+    # pull all args, excluding use_fast_accum flag if set.
+    A, B, A_scale, B_scale, bias, scale_result, out_dtype = args[:7]
+
+    assert A.dtype == torch.float8_e4m3fn
+    assert B.dtype == torch.float8_e4m3fn
+    # only cuSPARSELt supports float8_e4m3fn currentl
+    assert isinstance(A, torch.sparse.SparseSemiStructuredTensorCUSPARSELT)
+    assert A.packed is not None
+    # Currently we only support per-tensor scaling
+    assert A_scale.numel() == 1 and B_scale.numel() == 1
```

Comment:
For scaled mm we asset that the scale is fp32 regardless of our style, which is currently required by cubalaslt, It seems like if we pack into Alpha that is no longer a constraint?

Reply:
I think alpha must also be a float32 type, so this is still a requirement. I'll add an assert to check that scale dtypes is float32